cloud/terraform/aws_config_builder.py

```diff
@@ -35,13 +35,15 @@ def build_resources(self):
         self.resources_tf['resource']['aws_instance'] = {}
 
         for instance in self.resources_dict['instances']:
-            self.__new_aws_key_pair(instance['region'])
+            self.__new_aws_key_pair(instance)
             self.__new_aws_instance(instance)
 
         return self.resources_tf
 
-    def __new_aws_key_pair(self, region):
-        key_name = f'{region}-key{self.random_str}'
+    def __new_aws_key_pair(self, instance):
+        region = instance['region']
+        key_name = self.build_resource_name([instance['region'], 'key'])
```

Comment:
just being picky. Maybe `build` is not the best word for this function(?) What do you think of `create_resource_name`